### PR TITLE
Integration of Analytics in the Journey Builder

### PR DESCRIPTION
**Pull request by alfonsograziano**
Let's use this as reference for the complete integration: https://github.com/ogcio/life-events/pull/1028

### CODE REVIEW

File: apps/integrator/package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "@govie-ds/theme-govie": "^0.5.0",
     "auth": "*",
     "building-blocks-sdk": "*",
+    "@ogcio/building-blocks-sdk": "^0.0.19",
```
Comment (bozzelliandrea) [inline apps/integrator/package.json:27]: highly recommended to upgrade to the v21, we fixed the build error and added some additional features.

 just one breaking change about naming convention, for example
 `buildingBlocksSDK.analytics.trackEvent`
 becomes
 `buildingBlocksSDK.analytics.track.event`
 
 example [here](https://github.com/ogcio/life-events/blob/dfd85dbc8d9d410337f06c41695ed6e2c58096f7/apps/payments-api/routes/auditLogs/index.ts#L60)

File: apps/integrator/app/[locale]/(hosted)/journey/[journey]/run/[runId]/page.tsx [hunk 2]
```diff
@@ -38,15 +41,37 @@ export default async (props: Props) => {
...
+  if (!result?.url) {
     return (
       <div className="govie-width-container" style={{ width: "100%" }}>
         <div className="column">Something went wrong!</div>
       </div>
     );
   }
 
-  redirect(url, RedirectType.replace);
+  const analyticsSdk = await buildingBlocksSDK.analytics;
```
Comment (alewin) [inline apps/integrator/app/[locale]/(hosted)/journey/[journey]/run/[runId]/page.tsx:54]: `buildingBlocksSDK.analytics` is synchronous
```ts
const analyticsSdk = await buildingBlocksSDK.analytics;
```


so we can use it directly:
```ts
  await buildingBlocksSDK.analytics.track.event({
    event: {
      category: "StepExecution",
      action: "StepExecutionCompleted",
      name: result.stepType,
    },
    metadata: {
      userAgent: userAgent,
      language: locale,
    },
  });
```

File: apps/integrator/app/[locale]/(hosted)/journey/[journey]/run/[runId]/page.tsx [hunk 3]
```diff
@@ -38,15 +41,37 @@ export default async (props: Props) => {
...
+  await analyticsSdk.track.event({
+    event: {
+      category: "StepExecution",
+      action: "StepExecutionStarted",
+      name: result.stepType,
+    },
+    metadata: {
+      userAgent: userAgent,
+      language: locale,
+    },
```
Comment (alewin) [inline apps/integrator/app/[locale]/(hosted)/journey/[journey]/run/[runId]/page.tsx:73]: not important, but if you want there is the same method for metadata as well
```ts
buildingBlocksSDK.analytics.setTrackingMetadata({
      userAgent: userAgent,
      language: locale,
})
```

File: apps/integrator/libraries/analytics-auth-factory.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      trackingWebsiteId: process.env.ANALYTICS_WEBSITE_ID,
+    },
+  },
+  getTokenFn: getM2MTokenFn({
+    services: {
+      analytics: {
+        getOrganizationTokenParams: {
+          applicationId: process.env.M2M_INTG_ANALYTICS_APP_ID!,
+          applicationSecret: process.env.M2M_INTG_ANALYTICS_APP_SECRET!,
+          logtoOidcEndpoint: "http://localhost:3301/oidc", //process.env.LOGTO_ENDPOINT!,
```
Comment (alewin) [inline apps/integrator/libraries/analytics-auth-factory.ts:16]: why hardcoded?
```suggestion
          logtoOidcEndpoint: `${process.env.LOGTO_ENDPOINT}oidc`, 
```

Comment (nnorbert) [reply apps/integrator/libraries/analytics-auth-factory.ts:16]: This was a test only, I'll update it

**Review by alewin — DISMISSED** [review]

File: apps/integrator/package.json (hunk 1 above)
Comment (nnorbert) [reply apps/integrator/package.json]: Done!

File: apps/integrator/app/[locale]/(hosted)/journey/[journey]/run/[runId]/page.tsx (hunk 2 above)
Comment (nnorbert) [reply apps/integrator/app/[locale]/(hosted)/journey/[journey]/run/[runId]/page.tsx]: Done!

File: apps/integrator/app/[locale]/(hosted)/journey/[journey]/run/[runId]/page.tsx (hunk 3 above)
Comment (nnorbert) [reply apps/integrator/app/[locale]/(hosted)/journey/[journey]/run/[runId]/page.tsx:63]: Done!

File: apps/integrator/libraries/analytics-auth-factory.ts (hunk 4 above)
Comment (nnorbert) [reply apps/integrator/libraries/analytics-auth-factory.ts]: Done!

**Review by andreacappadona17 — APPROVED** [review]
Small comment on the website_id configuration, but it really depends on the config you have with the local API running

**Review by alewin — APPROVED** [review]

**Pull request merged**